pyqmc/dmc.py

```diff
@@ -292,68 +303,83 @@ def rundmc(
       weights: The final weights from this calculation
       
     """
-    # Restart
-    if hdf_file is not None:
-        with h5py.File(hdf_file, "a") as hdf:
-            if "configs" in hdf.keys():
-                stepoffset = hdf["step"][-1] + 1
-                configs.load_hdf(hdf)
-                weights = np.array(hdf["weights"])
-                if verbose:
-                    print("Restarted calculation")
+    # Restart from HDF file
+    if hdf_file is not None and os.path.isfile(hdf_file):
+        with h5py.File(hdf_file, "r") as hdf:
+            stepoffset = hdf["step"][-1] + 1
+            configs.load_hdf(hdf)
+            weights = np.array(hdf["weights"])
+            eref = hdf['eref'][-1]
+            esigma=hdf['esigma'][-1]
+            if verbose:
+                print("Restarted calculation")
+    else:
+        warmup = 2
+        df, configs = mc.vmc(wf, configs, accumulators=accumulators, client=client, npartitions=npartitions)
+        en = df[ekey[0]+ekey[1]][2:]
+        eref = np.mean(en).real
+        esigma = np.sqrt(np.var(en)*np.mean(df['nconfig']))
+        if verbose:
+            print("eref start", eref, "esigma", esigma)
 
     nconfig = configs.configs.shape[0]
     if weights is None:
         weights = np.ones(nconfig)
 
     npropagate = int(np.ceil(nsteps / branchtime))
     df = []
-
-    df_, configs, weights = propagate(
-        wf,
-        configs,
-        weights,
-        tstep,
-        branchcut_start=1e8,
-        branchcut_stop=1e9,
-        eref=0.0,
-        nsteps=1,
-        stepoffset=0,
-        accumulators=accumulators,
-        ekey=ekey,
-        drift_limiter=drift_limiter,
-        **kwargs,
-    )
-
-    df_ = pd.DataFrame(df_)
-    eref = df_[ekey[0] + ekey[1]][0].real
-    esigma = np.abs(eref) / 100
     for step in range(npropagate):
-        if verbose:
-            print("-", end='', flush=True)
-        df_, configs, weights = propagate(
-            wf,
-            configs,
-            weights,
-            tstep,
-            branchcut_start * esigma,
-            branchcut_stop * esigma,
-            eref=eref,
-            nsteps=branchtime,
-            stepoffset=branchtime * step + stepoffset,
```

Comment:
so now "step" in the hdf record is the average of all the steps between branching? the input kwarg nsteps is still npropagate * branchtime, not sure if that's confusing. but we probably don't need each individual step anyway, and the dmc trace will be smoother this way.

Reply:
yes, that might be confusing. I think this is the right structure, but the names might need to change.